File: languages/js/src/entity/identity_v2/decimal_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { AttributeDescriptor } from "./attribute_descriptor";

/**
 * Represents a descriptor for an attribute with a decimal value.
 */
export class DecimalAttributeDescriptor extends AttributeDescriptor {
  public toProto(): identityEntitiesProto.DecimalAttributeDefinitionV2 {
    return identityEntitiesProto.DecimalAttributeDefinitionV2.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.DecimalAttributeDefinitionV2
  ): DecimalAttributeDescriptor {
    return new DecimalAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required
    );
  }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity/DecimalAttributeDescriptor.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;

/**
 * Represents a descriptor for an attribute with a decimal value.
 */
public class DecimalAttributeDescriptor extends AttributeDescriptor {
  /**
   * Constructs an DecimalAttributeDescriptor object with the specified
   * parameters.
   * 
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   */
  public DecimalAttributeDescriptor(
      String displayName, String technicalName, String description, Boolean required) {
    super(displayName, technicalName, description, required);
  }

  public static DecimalAttributeDescriptor fromProto(
      BloockIdentityEntities.DecimalAttributeDefinition res) {
    return new DecimalAttributeDescriptor(
        res.getDisplayName(), res.getId(), res.getDescription(), res.getRequired());
  }

  public BloockIdentityEntities.DecimalAttributeDefinition toProto() {
    return BloockIdentityEntities.DecimalAttributeDefinition.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .build();
  }
}
